apps/mark-scan/backend/src/custom-paper-handler/state_machine.ts

```diff
@@ -372,20 +372,29 @@ export function buildMachine(
         },
       },
       printing_ballot: {
-        invoke: pollPaperStatus(),
-        entry: (context, event) => {
-          // Need to hint to Typescript that we want the 'VOTER_INITIATED_PRINT' event in our union type of events
-          if ('pdfData' in event) {
-            void driverPrintBallot(context.driver, event.pdfData, {});
-          } else {
-            throw new Error(
-              `printing_ballot entry called by unsupported event type: ${event.type}`
-            );
-          }
-        },
-        on: {
-          PAPER_IN_OUTPUT: 'scanning',
-        },
+        invoke: [
+          {
+            id: 'printBallot',
+            src: (context, event) => {
+              // Need to hint to Typescript that we want the 'VOTER_INITIATED_PRINT' event in our union type of events
+              if ('pdfData' in event) {
```

Comment:
You could do `assert(event.type === 'VOTER_INITIATED_PRINT')` to make this a bit more concise. Our `assert` function will hint the type checker

Reply:
[723c854](https://github.com/votingworks/vxsuite/pull/3936/commits/723c854a1a00a702f440201a0feb2151f90fc49b)